src/applications/model/gaming-streaming-server.h

```diff
@@ -96,32 +101,40 @@ class GamingStreamingServer : public Application
    */
   void AddNewTraffic (Ptr<RandomVariableStream> packetSize, Ptr<RandomVariableStream> interArrivalTime);
 
+  virtual void StartApplication (void);
+  virtual void StopApplication (void);
+
+protected:
   /**
-   * \brief Generate and send packets based on the different distribution
+   * \brief Generate and send packets based on the different random distribution
    *
-   * \param packetSize random variable which defines the size of generated packets
-   * \param interArrivalTime random variable which defines inter-arrival times between the packets
+   * \param idx index of stream within the streams vector
    */
-  void Send (Ptr<RandomVariableStream> packetSize, Ptr<RandomVariableStream> interArrivalTime);
+  void Send (int idx);
 
   /**
    *  Initialize the parameters of different streams
    */
   virtual void InitializeStreams (void);
 
 private:
+  /**
+   * \brief Schedule the next packet transmission
+   *
+   * \param idx index of stream
+   */
+  void ScheduleNextTx (int idx);
 
   uint32_t     m_seq;  //!< Sequence number for packets
   uint32_t     m_totSentPackets; //!< Counter for sent packets
+  uint32_t     m_totSentBytes; //!< Total bytes sent so far
   Ptr<Socket>  m_socket; //!< Socket
   Address      m_peerAddress; //!< Remote peer address
   uint16_t     m_peerPort; //!< Remote peer port
-  //TODO: std::vector<RandomVariableStream> is sufficient???
-  std::vector <Ptr<RandomVariableStream> > m_pksList;  //!< A list of random number generator for packet size
-  std::vector <Ptr<RandomVariableStream> > m_iatList; //!< A list of random number generator for packets inter-arrival time
+  std::vector<EventId> m_sendEvent; //!< Sent event for next packet in each streams
 
-  virtual void StartApplication (void);
-  virtual void StopApplication (void);
+  std::vector<Ptr<RandomVariableStream> > m_packetSizeVariables;  //!< A list of random number generator for packet size
+  std::vector<Ptr<RandomVariableStream> > m_interArrivalTimesVariables; //!< A list of random number generator for packets inter-arrival time
```

Comment:
given their strong inter-dependence we could even define a
```cpp
struct TrafficStream:
{
  EventId m_sendEvent;  //!< Send event for next packet
  Ptr<RandomVariableStream> m_packetSizeVariable;  //!< Random number generator for packet size
  Ptr<RandomVariableStream> m_interArrivalTimesVariable; //!< Random number generator for packets inter-arrival time
}

std::vector<TrafficStream> m_trafficStreams; //!< A list of traffic streams
```
what do you think @salmanmohebi @MatteoDrago @tommyazz ?

Reply:
Yes I agree, I would be clearer like this